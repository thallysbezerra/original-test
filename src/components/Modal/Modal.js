import React from 'react';
import PropTypes from 'prop-types';

import './Modal.scss';

const Modal = ({ children, onClickCloseModal, modalIsActive }) => (
	<div className={`modal--shadow ${modalIsActive ? 'active' : ''}`}>
		<div className={`modal ${modalIsActive ? 'active' : ''}`}>
			<button className="modal__button--close" onClick={onClickCloseModal}>
				<i className="fas fa-close" />
			</button>
			{children}
		</div>
	</div>
);

Modal.propTypes = {
	showModal: PropTypes.bool.isRequired
};

Modal.defaultProps = {
	showModal: false
};

export default Modal;
