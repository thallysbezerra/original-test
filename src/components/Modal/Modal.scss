@import '../../css/Theme.scss';

.modal {
	background-color: $white;
	border-radius: 3px;
	box-sizing: border-box;
	height: auto;
	left: 50%;
	padding: 60px 20px 0;
	position: fixed;
	top: 50%;
	width: 80%;
	transform: translate(-50%, 50%);
	transition: transform 0.25s ease-in-out;
	z-index: 2;

	&.active {
		transform: translate(-50%, -50%);
	}

	@include breakpoint(tabletL) {
		max-height: 600px;
		max-width: 445px;
	}

	&--shadow {
		background-color: rgba(32, 32, 32, 0.6);
		height: 100%;
		left: 0;
		opacity: 0;
		position: fixed;
		top: 0;
		transition: opacity 0.25s ease-in-out, visibility 0.25s ease-in-out;
		visibility: hidden;
		width: 100%;
		z-index: 1;

		&.active {
			opacity: 1;
			visibility: visible;
		}
	}

	&__button {
		&--close {
			background-color: transparent;
			border: none;
			font-size: 24px;
			position: absolute;
			right: 10px;
			top: 10px;

			&:hover {
				cursor: pointer;
			}
		}

		&--keep-buying {
			margin: 10px 0;
		}
	}

	&__img {
		display: block;
		margin: 0 auto;
		max-width: 260px;
	}

	&__msg {
		color: $salmon;
		font-family: 'Gotham Bold';
		font-size: 22px;
		margin: 10px 0;
		text-transform: uppercase;
	}
}
