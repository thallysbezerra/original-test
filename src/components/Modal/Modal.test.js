import React from 'react';
import Modal from './Modal';

import renderer from 'react-test-renderer';

describe('Modal UI', () => {
	it('<Modal /> must have the snapshot version.', () => {
		const tree = renderer.create(<Modal />);

		expect(tree.toJSON()).toMatchSnapshot();
	});
});
