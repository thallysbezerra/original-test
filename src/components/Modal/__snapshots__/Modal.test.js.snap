// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal UI <Modal /> must have the snapshot version. 1`] = `
<div
  className="modal--shadow "
>
  <div
    className="modal "
  >
    <button
      className="modal__button--close"
    >
      <i
        className="fas fa-close"
      />
    </button>
  </div>
</div>
`;
